cms/src/components/content/query.ts

```diff
@@ -0,0 +1,131 @@
+import { db, DocType, TagType, type Uuid, type ContentDto, PublishStatus } from "luminary-shared";
+
+export type ContentOverviewQueryOptions = {
+    languageId: Uuid;
+    parentType: DocType.Post | DocType.Tag;
+    tagType?: TagType;
+    orderBy?: "title" | "updatedTimeUtc" | "publishDate" | "expiryDate";
+    orderDirection?: "asc" | "desc";
+    translationStatus?: "translated" | "untranslated" | "all";
+    publishStatus?: "published" | "scheduled" | "expired" | "draft" | "all";
+    pageSize?: number;
+    pageIndex?: number;
+    tags?: Uuid[];
+    search?: string;
+};
+
+async function contentOverviewQuery(options: ContentOverviewQueryOptions) {
+    if (!options.orderBy) options.orderBy = "updatedTimeUtc";
+    if (!options.orderDirection) options.orderDirection = "desc";
+    if (!options.pageSize) options.pageSize = 20;
+    if (!options.pageIndex) options.pageIndex = 0;
+    if (!options.translationStatus) options.translationStatus = "all";
+    if (!options.publishStatus) options.publishStatus = "all";
+
+    const translated = (await db.docs // This may slow down the query if there are many documents, but it is necessary to be able to include and filter on untranslated documents
+        .where({ type: DocType.Content, language: options.languageId })
+        .toArray()) as ContentDto[];
+    const untranslatedByParentId: Uuid[] = [];
+
+    let res = db.docs.orderBy(options.orderBy);
```

Comment:
Shouldn't this have a `where` for only Content docs? Now we're loading everything.

I also guess this is where pagination will come into play? As this can load a very large dataset.

Reply:
The where includes ```{ type: DocType.Content, language: options.languageId }```, but I suppose I might have added it after your comment.

Reply:
That is the other query for `translated` it seems?